### ISSUE TITLE
Reproducibility results

### ISSUE THREAD

**Issue by HamidGadirov** [megvii-research/ECCV2022-RIFE#297]
Hi,

I checked if I can reproduce the results similar to those in the paper to make sure I am training the model properly. These are the results that I got on Vimeo triplets:

<img width="1498" alt="interpol_flow" src="https://user-images.githubusercontent.com/39424882/207704634-2260fa01-98dd-4800-ae39-edb410cab2ef.png">

The model prediction is shown for t=1 (predicting between t=0 and t=2), the second row corresponds to interpolation ("Interpol"), and the last row to flow ("Flow pred"). I see that interpolation results are very good, however, I expected the flow to be a bit more accurate. In section 6.2 of the appendix you mention that "IFNet produces clear motion boundaries.", this is also what can be seen in Figure 10. Therefore I wanted to ask if there were any other training steps that I need to add to get flow prediction more accurate. I can of course share more prediction examples that I got.

The training was done for 300 epochs using the reconstruction losses and distillation loss (the latter is with coeff. 0.01) as described in the paper, I didn't change anything in the code to train and obtain these results. This is the loss plot:
![val_loss_300](https://user-images.githubusercontent.com/39424882/207704725-822e8e51-3a4b-469e-bc00-f86a59c881b0.png)

I would appreciate if you can confirm that you trained the model the same way and that your flow predictions look similar. I used IFNet for training (`self.flownet = IFNet()`).

**Comment by hzwer** [comment]
I don't know how to make the optical flow result better for the time being. I want to say that I actually don't know how to define a "good optical flow result"

My visualization is as follows：

![image](https://user-images.githubusercontent.com/10103856/208048531-c7fe0d34-6207-4a4b-8449-3fb9c2db57f1.png)
![image](https://user-images.githubusercontent.com/10103856/208048540-5cdf2ffe-4b54-41f2-9217-140cdd83c9f3.png)
![image](https://user-images.githubusercontent.com/10103856/208048550-b3e4c6e8-7431-45e9-b644-f5cbec60fdea.png)

**Comment by HamidGadirov** [comment]
Thanks for sharing the flow results. I tried to improve them on my datasets using regularization, smoothness, and photometric loss components. From my experiments, I see some improvement in terms of predicted flow with regularization and photometric constraints, that e.g. fixes the flow direction and penalizes unnecessary flow vectors. The interpolation is very good in all scenarios once the model converges properly. Did you try earlier to add e.g. photometric loss in your experiments? It might be more challenging to get the network to converge when using more loss components.

Also, I saw in the previous discussion ("training questions" issue) that you are not using distillation loss anymore, but instead have VGG loss. I guess it will give some improvement in terms of interpolation, but I am not sure if that can help for the optical flow prediction. Is the code for the new version  (with only L1 + VGG loss) available somewhere?

**Comment by hzwer** [comment]
Have you heard of TOFlow? https://github.com/anchen1011/toflow
Because I believe that optical flow is just a description of a type of pixel relationship, and for real-time video, optical flow is ambiguous. When downstream tasks are considered, the optical flow properties change again. If you're after better optical flow, maybe work in the field of unsupervised flow can help.

**Comment by HamidGadirov** [comment]
I didn't hear of TOFlow, I checked the paper but they pre-train their motion estimation network on Sintel, in a supervised mode.

I already tried to utilize e.g. UPFlow (https://github.com/coolbeam/UPFlow_pytorch) but I faced the issue that their training was specific for the CV datasets and there were several training stages involved. I wasn't able to reproduce the results easily and actually my research is to apply such techniques to different type of datasets, so I decided to return to RIFE and improve its results. I am also thinking about adding supervised loss to existing reconstruction and distillation losses to improve the flow, other experiments involved adding regularization and photometric components, as I mentioned earlier. I am trying to get consistent results on all datasets at the moment.
Sometimes I saw that distillation error becomes NaN, I may try Batch normalization as you suggested.
